src/lib/sap_ha/configuration/hana.rb

```diff
@@ -286,35 +332,38 @@ def config_firewall(role)
 
       # Creates the sudoers file
       def adapt_sudoers
-	if File.exist?(SapHA::Helpers.data_file_path("SUDOERS_HANASR.erb"))
-	  Helpers.write_file("/etc/sudoers.d/saphanasr.conf",Helpers.render_template("SUDOERS_HANASR.erb", binding))
-	end
+        if File.exist?(SapHA::Helpers.data_file_path("SUDOERS_HANASR.erb"))
+          Helpers.write_file("/etc/sudoers.d/saphanasr.conf",Helpers.render_template("SUDOERS_HANASR.erb", binding))
+        end
       end
 
       # Activates all necessary plugins based on role an scenario
       def adjust_global_ini(role)
         # SAPHanaSR is needed on all nodes
-        add_plugin_to_global_ini("SAPHANA_SR")
+        add_plugin_to_global_ini("SAPHANA_SR", @system_id)
         if @additional_instance
           # cost optimized
-          add_plugin_to_global_ini("SUS_COSTOPT") if role != :master
+          add_plugin_to_global_ini("SUS_COSTOPT", @system_id) if role != :master
+          add_plugin_to_global_ini("NON_PROD", @np_system_id) if role != :master
+          command = ["hdbnsutil", "-reloadHADRProviders"]
+          _out, _status = su_exec_outerr_status("#{@np_system_id.downcase}adm", *command)
         else
           # performance optimized
-          add_plugin_to_global_ini("SUS_CHKSRV")
-          add_plugin_to_global_ini("SUS_TKOVER")
+          add_plugin_to_global_ini("SUS_CHKSRV", @system_id)
+          add_plugin_to_global_ini("SUS_TKOVER", @system_id)
         end
         command = ["hdbnsutil", "-reloadHADRProviders"]
-        out, status = su_exec_outerr_status("#{@system_id.downcase}adm", *command)
+        _out, _status = su_exec_outerr_status("#{@system_id.downcase}adm", *command)
```

Comment:
well, ruby is not strict with not using return values. So if you do not need it, just do not assign it
```suggestion
        su_exec_outerr_status("#{@system_id.downcase}adm", *command)
```